clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -182,14 +202,17 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
 
     @Override
     public void commitTransaction() throws ProducerFencedException {
-        if (producerFencedOnCommitTxn) {
-            throw new ProducerFencedException("Producer is fenced");
-        }
-
```

Comment:
I think we should do the regular state check first before we throw any test ingested exception.

Reply:
We just replaced this with the `commitTransactionException` -- a user can just set a `ProducerFencedException` to get the same behavior